nshlib/nsh_consolemain.c

```diff
@@ -65,6 +65,10 @@ int nsh_consolemain(int argc, FAR char *argv[])
   int ret;
 
   DEBUGASSERT(pstate != NULL);
+  if (pstate == NULL)
```

Comment:
Should we remove `DEBUGASSERT` above?

Reply:
Good question. It would change the behavior in debug profile and if you relied on the debugassert and didn't check the return value, it would then silently fail.
I personally would leave it in, even if it is duplicated code in debug, just to maintain the existing behavior.